src/Analyzers/MSTest.Analyzers/ClassInitializeShouldBeValidAnalyzer.cs

```diff
@@ -75,6 +80,38 @@ public override void Initialize(AnalysisContext context)
             return;
         }
 
+        if (namedTypeSymbol.IsGenericType)
```

Comment:
My previous feedback was closed, but I still think this can move up higher.  IsClassInitializeMethod checks all the attributes, but if the type is not generic we don't care about any method on it. (at least that is how I understand the rule.

Reply:
the test class doesn't have to be generic to have a classInit method but the rule is
in case the test class is generic and has a classinit method it should have the inheritance behaviour 

so based on that I think we don't to change 
@nohwnd what do you think?

Reply:
Yes that is how I understand it as well. 

if class is NOT generic, this rule does not apply at all, and can return immediately. No matter if the method we are inspecting is classInit.

So to me this makes sense to change, because we can return immediately with a cheaper check (class.isGeneric), rather than later (method.isClassInit && class.IsGeneric)

Reply:
the problem is if it's a generic class, I can't Report Diagnostic before making sure first that's a classInit method

Reply:
I did not notice this rule is implementing other checks to make sure a class initialize method is valid. In that case returning right away would not make sense.